packages/@lwc/engine-core/src/framework/check-version-mismatch.ts

```diff
@@ -38,7 +38,7 @@ export function checkVersionMismatch(
 ) {
     const versionMatcher = func.toString().match(LWC_VERSION_COMMENT_REGEX);
     if (!isNull(versionMatcher) && !warned) {
-        if (process.env.SKIP_LWC_VERSION_MISMATCH_CHECK === 'true') {
+        if (typeof process === 'object' && process.env.SKIP_LWC_VERSION_MISMATCH_CHECK === 'true') {
```

Comment:
```suggestion
        if (!process.env.IS_BROWSER && process.env.SKIP_LWC_VERSION_MISMATCH_CHECK === 'true') {
```
We should check `IS_BROWSER` here, because that gets hard-coded by rollup, meaning the whole block can be tree shook when appropriate. (It's also helpful for future maintainers to see our intentions, here.)